packages/webpack-plugin/src/plugin-utils.ts

```diff
@@ -0,0 +1,368 @@
+        .map((m) => {
+            return replaceMappedCSSAssetPlaceholders({
+                assetsModules: assetsModules,
+                staticPublicPath,
+                chunkGraph,
+                moduleGraph,
+                dependencyTemplates,
+                runtime,
+                runtimeTemplate,
+                stylableBuildMeta: m.buildMeta.stylable,
+            });
+        });
+
+    return cssChunks;
+}
+
+export function getStylableBuildMeta(module: Module): StylableBuildMeta {
+    const meta = module.buildMeta.stylable;
+    if (!meta) {
+        throw new Error('Stylable module does not contains build meta');
+    }
+    return meta;
+}
+
+export function findIfStylableModuleUsed(m: Module, compilation: Compilation) {
+    const moduleGraph = compilation.moduleGraph;
+    const chunkGraph = compilation.chunkGraph!;
+    const inConnections = uniqueFilterMap(
+        moduleGraph.getIncomingConnections(m),
+        ({ resolvedOriginModule, dependency }) =>
+            dependency instanceof UnusedDependency ? undefined : resolvedOriginModule
+    );
+
+    // TODO: check if this optimization is good.
+    // const inChunks = chunkGraph.getNumberOfModuleChunks(m)
+    // if(inChunks === 0) {
+    //     debugger
+    // }
+
+    let isInUse = false;
+    for (const cm of inConnections) {
+        if (cm.buildMeta.sideEffectFree) {
+            const info = moduleGraph.getExportsInfo(cm);
+            const usedExports = (info.getUsedExports as any)(/*if passed undefined it finds usages in all chunks*/);
+            if (usedExports === false) {
+                continue;
+            } else if (usedExports === true || usedExports === null) {
+                /** noop */
+            } else if (usedExports.size === 0) {
+                continue;
+            }
+        }
+        const chunksCount = chunkGraph.getNumberOfModuleChunks(cm);
+
+        if (chunksCount > 0) {
+            isInUse = true;
+            break;
+        }
+    }
+    return isInUse;
+}
+
+export function getFileName(filename: string, data: Record<string, string>) {
+    return filename.replace(/\[(.*?)]/g, (fullMatch, inner) => {
+        const [type, len] = inner.split(':');
+        const value = data[type];
+        if (value) {
+            const length = Number(len);
+            return !isNaN(length) ? value.slice(0, length) : value;
+        } else {
+            return fullMatch;
+        }
+    });
+}
+
+export function getSortedModules(stylableModules: Set<NormalModule>) {
+    return Array.from(stylableModules)
+        .sort((m1, m2) => {
+            if (m1.resource > m2.resource) {
+                return 1;
+            } else if (m1.resource < m2.resource) {
+                return -1;
+            } else {
+                return 0;
+            }
+        })
+        .sort((m1, m2) => getStylableBuildMeta(m2).depth - getStylableBuildMeta(m1).depth);
```

Comment:
.sort can be called once, and when depthA - depthB === 0, fallback to string sorting

Reply:
done